bundles/binding/org.openhab.binding.km200/src/main/java/org/openhab/binding/km200/internal/KM200Comm.java

```diff
@@ -0,0 +1,712 @@
+            type = object.getServiceType();
+        } else {
+            logger.error("Service is not in the determined device service list: " + service);
+            return;
+        }
+
+        logger.debug("state of: " + command.toString() + "type: " + type.toString());
+        if (itemType.isAssignableFrom(NumberItem.class)) {
+            Float val = ((DecimalType) command).floatValue();
+            if (object.getValueParameter() != null) {
+                List<Float> valParas = (List<Float>) object.getValueParameter();
+                Float minVal = valParas.get(0);
+                Float maxVal = valParas.get(1);
+                if (val < minVal) {
+                    val = minVal;
+                }
+                if (val > maxVal) {
+                    val = maxVal;
+                }
+            }
+            if (type.equals("floatValue")) {
+                nodeRoot = new JSONObject().put("value", val);
+            } else if (type.equals("stringValue")) {
+                nodeRoot = new JSONObject().put("value", val.toString());
+            } else {
+                logger.error("Not supported type for numberItem:" + type.toString());
+            }
+        } else if (itemType.isAssignableFrom(StringItem.class)) {
+            String val = ((StringItem) command).toString();
+            if (object.getValueParameter() != null) {
+                List<String> valParas = (List<String>) object.getValueParameter();
+                if (!valParas.contains(val)) {
+                    throw new IllegalArgumentException("Parameter is not in the service parameterlist:" + val);
+                }
+            }
+            if (type.equals("stringValue")) {
+                nodeRoot = new JSONObject().put("value", val);
+            } else if (type.equals("floatValue")) {
+                nodeRoot = new JSONObject().put("value", Float.parseFloat(val));
+            } else {
+                logger.error("Not supported type for stringItem:" + type.toString());
+            }
+
+        } else if (itemType.isAssignableFrom(DateTimeItem.class)) {
+            String val = ((DateTimeItem) command).toString();
+            if (type.equals("stringValue")) {
+                nodeRoot = new JSONObject().put("value", val);
+            } else {
+                logger.error("Not supported type for dateTimeItem:" + type.toString());
+            }
+
+        } else if (itemType.isAssignableFrom(SwitchItem.class)) {
+            String val = null;
+            if (provider.getParameter(item).containsKey("on")) {
+                if (command == OnOffType.OFF) {
+                    val = provider.getParameter(item).get("off");
+                } else if (command == OnOffType.ON) {
+                    val = provider.getParameter(item).get("on");
+                }
+            } else {
+                logger.error("witch-Item only on configured on/off string values:" + command.toString());
```

Comment:
"Switch-Item"

Also, use parameterized logging instead of string concatenation